lib/Bindings/Python/circt/support.py

```diff
@@ -129,14 +133,28 @@ def attribute_to_var(attr):
   raise TypeError(f"Cannot convert {repr(attr)} to python value")
 
 
+def get_self_or_inner(mlir_type):
```

Comment:
I was going to propose a C API for our type_isa, type_cast, type_dyn_cast functionality rather than doing something separate in Python. But now that I remember the fun that necessitates `type_to_pytype`, I'm less sure such a C API and Python binding would make this any better...

Reply:
Yeah, plus remember our experience with trying to re-use the C++ backedge builder? I've given up on trying to re-use C++ through the C API.